api/resolvers/wallet.js

```diff
@@ -54,20 +54,25 @@ function injectResolvers (resolvers) {
         settings && Object.keys(validData).filter(key => key in settings).forEach(key => { settings[key] = validData[key] })
       }
 
+      // wallet in shape of db row
+      const wallet = {
+        field: walletDef.walletField,
+        type: walletDef.walletType,
+        userId: me?.id
+      }
+      const logger = walletLogger({ wallet, models })
+
       return await upsertWallet({
-        wallet: {
-          field: walletDef.walletField,
-          type: walletDef.walletType
-        },
+        wallet,
         testCreateInvoice:
           walletDef.testCreateInvoice && validateLightning && canReceive({ def: walletDef, config: data })
-            ? (data) => walletDef.testCreateInvoice(data, { me, models })
+            ? (data) => walletDef.testCreateInvoice(data, { logger, me, models })
```

Comment:
we pass `logger` to `testCreateInvoice` of the wallets but we only use the logger for NWC. So far, I saw no need to add wallet-specific logging to the other wallets.